### ISSUE TITLE
nixUnstable  1.12pre5344_eba840c8 fails to build

### ISSUE THREAD

**Issue by nlewo** [NixOS/nixpkgs#25607]
## Issue description
nixUnstable fails to build on master since d53540d

```
$ nix-build ./ -A nixUnstable
...
...
...
+ [[ ! /nix/store/wsz11sx18n85if6gp50m870rypgavwpd-bash-4.4-p12/bin/bash =~ /nix/store ]]
+ chmod -R u+w /tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store0
+ chmod -R u+w /tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store1
chmod: cannot access '/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store1': No such file or directory
+ true
+ rm -rf /tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store0 /tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store1
+ export NIX_CONF_DIR=/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/etc2
+ NIX_CONF_DIR=/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/etc2
+ mkdir -p /tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/etc2
+ echo 'build-sandbox-paths = /nix/store'
++ nix-build build-hook.nix --no-out-link -j0 --option builders 'local?root=/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store0; local?root=/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store1 - - 1 1 foo' --option build-sandbox-paths /nix/store
these derivations will be built:
  /tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store/cn77i60jq3rbyf0r3im3h2j2fhh4x6dq-build-hook-input-2.drv
  /tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store/z5is0w3pbcwfv5kyv6df8j25pqmnvzzv-build-hook-input-1.drv
  /tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store/8mpd0nil17l2mjfgbw0ypy9g2gn0x46z-build-hook.drv
warning: unknown setting 'log-fd'
warning: unknown setting 'max-connections'
warning: unknown setting 'log-fd'
warning: unknown setting 'max-connections'
copying ‘/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store/cj0ki8zva2hlmg0f69fv6ycknbcdv6rv-dependencies.builder1.sh’...
building ‘/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store/z5is0w3pbcwfv5kyv6df8j25pqmnvzzv-build-hook-input-1.drv’ on ‘local?root=/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store1’
building path(s) ‘/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store/3j32wbdxaa8gpzg19lb4cmv91sgn6mmk-build-hook-input-1’
error: path ‘/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store/3j32wbdxaa8gpzg19lb4cmv91sgn6mmk-build-hook-input-1’ is not valid
cannot build derivation ‘/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store/8mpd0nil17l2mjfgbw0ypy9g2gn0x46z-build-hook.drv’: 1 dependencies couldn't be built
error: build of ‘/tmp/nix-build-nix-1.12pre5344_eba840c8.drv-0/nix-test/store/8mpd0nil17l2mjfgbw0ypy9g2gn0x46z-build-hook.drv’ failed
+ outPath=
FAIL: tests/build-remote.sh
2 out of 41 tests failed 
make: *** [mk/tests.mk:11: installcheck] Error 1
builder for ‘/nix/store/ams289nf0p3v05wy5vk91fd7dps6q37v-nix-1.12pre5344_eba840c8.drv’ failed with exit code 2
error: build of ‘/nix/store/ams289nf0p3v05wy5vk91fd7dps6q37v-nix-1.12pre5344_eba840c8.drv’ failed
```

### Steps to reproduce
`$ nix-build ./ -A nixUnstable`

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
16.09.1942.28dc5c7d22
* Nix version: (run `nix-env --version`)
nix-env (Nix) 1.11.8
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
16.09.1942.28dc5c7d22
* Sandboxing enabled: (run `grep build-use-sandbox /etc/nix/nix.conf`)

**Comment by globin** [comment]
ping @edolstra

**Comment by edolstra** [comment]
This is probably because your kernel is too old (< 4.6 I think).

I'm working on a fix for older kernels.

**Comment by copumpkin** [comment]
Unrelated to the issue at hand, but:

> warning: unknown setting 'log-fd'
> warning: unknown setting 'max-connections'

😍😍

**Comment by edolstra** [comment]
Fixed in 234cdc02eb.

**Comment by nlewo** [comment]
Thx
